rpc/src/transaction_notifier_interface.rs

```diff
@@ -0,0 +1,17 @@
+use {
+    solana_sdk::{clock::Slot, signature::Signature, transaction::SanitizedTransaction},
+    solana_transaction_status::TransactionStatusMeta,
+    std::sync::{Arc, RwLock},
+};
+
+pub trait TransactionNotifierInterface {
+    fn notify_transaction(
+        &self,
+        slot: Slot,
+        signature: &Signature,
+        transaction_status_meta: &TransactionStatusMeta,
+        transaction: &SanitizedTransaction,
+    );
+}
+
+pub type TransactionNotifier = Arc<RwLock<dyn TransactionNotifierInterface + Sync + Send>>;
```

Comment:
I think I prefer to be more explicit:

1. TransactionNotifierInterface -> TransactinNotifier
2. TransactionNotifier -> TransactionNotifierLock